api/src/main/java/org/openmrs/BaseOpenmrsData.java

```diff
@@ -90,15 +90,15 @@ public void setCreator(User creator) {
 	 */
 	@Override
 	public Date getDateCreated() {
-		return dateCreated;
+		return dateCreated != null ? new Date(dateCreated.getTime()) : null;
```

Comment:
@dkayiwa @mogoodrich  Whats your opinion on always creating a new date when calling the getDate method ? i have seen this in lots of pull requests lately because of the sonar warning. In my opinion  get method should just return the original instance of the date object rather than creating a new date instance each time

Reply:
What do you think of this? http://www.informit.com/articles/article.aspx?p=31551&seqNum=2

Reply:
For a immutable object it make sense. but this class is not immutable. Its providing a setter method for the date object .  Well this is just my thought on this. If you are fine with this im ok too..

Reply:
My apologies, I am not very familiar with the code base and was not aware that the class isn't mutable.